### DISCUSSION TITLE
How to use multiple fonts in multi_cell if it contains english and arabic letters

### DISCUSSION THREAD

**Discussion by papunmohanty-homer** [py-pdf/fpdf2#583] [Q&A]
If I need to print mixed fonts how can I achieve it
`pdf.multi_cell(w=150, h=5, txt="تمانتمانتمانتمانتم English Letters") `
I have posted my question in Stackoverflow as well
https://stackoverflow.com/questions/74059846/how-to-print-arabic-and-english-text-together-using-pythons-fpdf-library

**Comment by Lucas-C** [comment]
Hi @papunmohanty-homer and welcome!

Interesting question.
Currently, you will need to make several calls to `multi_cell()`:
```python
from fpdf import FPDF

pdf = FPDF()
pdf.add_font("NotoSansArabic", fname="NotoSansArabic.ttf")
pdf.add_page()

pdf.set_font("NotoSansArabic", size=24)
text = "تمانتمانتمانتمانتم"
text_width = pdf.get_string_width(text)
pdf.multi_cell(w=text_width + 2, h=5, txt=text)

pdf.set_font("Helvetica", size=24)
pdf.multi_cell(w=0, h=5, txt="English Letters")

pdf.output("discuss_583.pdf")
```

Does that solve your issue @papunmohanty-homer?

**Reply by papunmohanty-homer** [reply]
Yes, calling `multi_cell()` is working
here is the sample I am able to produce
Many thanks for helping out
![image](https://user-images.githubusercontent.com/93320729/196121442-1bd0f4a7-f0d4-46ef-b8a6-69d66b9405ce.png)

**Reply by Lucas-C** [reply]
Great!

**Reply by Lucas-C** [reply]
Another alternative is to use [write_html()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.write_html)
```python
from fpdf import FPDF, HTMLMixin

class PDF(FPDF, HTMLMixin):
        pass

pdf = PDF()
pdf.add_font("NotoSansArabic", fname="NotoSansArabic.ttf")
pdf.add_page()
pdf.write_html("""
  <font face="NotoSansArabic">تمانتمانتمانتمانتم</font>
  <font face="helvetica">English Letters</font>
""")
pdf.output("dual-text.pdf")
```

Note that `HTMLMixin` will be deprecated in the next release, and `.write_html` will be natively available in `FPDF`:
https://github.com/PyFPDF/fpdf2/blob/master/CHANGELOG.md#deprecated

**Comment by gmischler** [comment]
Apparently, the Noto Sans Arabic font does not include any latin character glyphs. If you absolutely need to use this specific font, then the suggestion by @Lucas-C above is the way to go (or a variation using the `.write()` method so you don't need to do all the positioning manually).
If you can chose your font freely, then you might want to find one that includes glyphs for both the arabic and latin scripts, and your problem will magically go away...
